homeassistant/components/samsungtv/media_player.py

```diff
@@ -158,12 +162,21 @@ def update(self) -> None:
         else:
             self._attr_state = STATE_ON if self._bridge.is_on() else STATE_OFF
 
-    def send_key(self, key: str) -> None:
+        if self._attr_state == STATE_ON and self._app_list is None:
+            self._app_list = {}  # Ensure that we don't update it twice in parallel
+            self.hass.async_add_job(self._update_app_list)
+
+    def _update_app_list(self) -> None:
+        self._app_list = self._bridge.get_app_list()
+        if self._app_list is not None:
+            self._attr_source_list.extend(self._app_list)
```

Comment:
You're extending this list each time the TV turns on. Instead you should reset it first to `list(SOURCES)` before extending it.